src/macros.rs

```diff
@@ -27,76 +27,57 @@ use crate::CStr8;
 /// ```
 #[macro_export]
 macro_rules! cstr8 {
-    ($s:expr) => {
-        unsafe {
-            const __CSTR: [$crate::u8; $crate::cstr_len_for($s)] = unsafe {
-                $crate::transmute_prefix($crate::Concat(
-                    $crate::str_as_byte_array::<{ $crate::str_len($s) }>($s),
-                    b'\0',
-                ))
-            };
-            $crate::CStr8::from_utf8_with_nul_unchecked(&__CSTR)
-        }
-    };
+    ($s:expr) => {{
+        const CSTR: $crate::__internal_unstable::CStr8Array<
```

Comment:
FYI, the reason I renamed the `const` item back to `__CSTR` is that while local binding names are hygienic in `macro_rules!`, *item* names aren't. Thus using `__CSTR` as the name avoids causing issues when the user has a `CSTR` name already in scope (and potentially uses it in `$s`). Double underscore doesn't *prevent* such issues, but it makes them significantly less likely to happen accidentally.

Reply:
Ah, you're right, I was depending on old (probably always incorrect) info: https://danielkeep.github.io/tlborm/book/mbe-min-hygiene.html